readthedocs/restapi/views/model_views.py

```diff
@@ -205,13 +225,23 @@ class BuildViewSetBase(UserSelectViewSet):
     admin_serializer_class = BuildAdminSerializer
     model = Build
 
+    def log(self, request, *args, **kwargs):
+        build = self.get_object()
+        return Response(build.raw_log)
+
 
 class BuildViewSet(SettingsOverrideObject):
 
     """A pluggable class to allow for build cold storage."""
 
     _default_class = BuildViewSetBase
 
+    @detail_route(renderer_classes=[PlainTextRenderer])
+    def log(self, request, *args, **kwargs):
```

Comment:
You don't need to redefine this here, but you need to move the `@detail_route` decorator to the definition in the `Base` class.

Reply:
I try that, but was unsuccessful, here the decorator gets exposed registering the function, then `SettingsOverrideObject` call this function from the class on `_default_class`.

Reply:
Mmm... I don't know exactly how the `SettingsOverrideObject` works but maybe @agjohnson can help us here. Anyway, I think this is not the _correct_ way of doing it and I'm not sure if this won't cause a problem in the .com site (the reason why we need the override object)